flow/connectors/clickhouse/qrep_avro_sync.go

```diff
@@ -155,13 +157,36 @@ func (s *ClickHouseAvroSyncMethod) SyncQRepRecords(
 	if creds.AWS.SessionToken != "" {
 		sessionTokenPart = fmt.Sprintf(", '%s'", creds.AWS.SessionToken)
 	}
-	query := fmt.Sprintf("INSERT INTO `%s`(%s) SELECT %s FROM s3('%s','%s','%s'%s, 'Avro')",
-		config.DestinationTableIdentifier, selectorStr, selectorStr, avroFileUrl,
-		creds.AWS.AccessKeyID, creds.AWS.SecretAccessKey, sessionTokenPart)
 
-	if err := s.database.Exec(ctx, query); err != nil {
-		s.logger.Error("Failed to insert into select for ClickHouse", slog.Any("error", err))
-		return 0, err
+	hashColName := dstTableSchema[0].Name()
+	numParts, err := peerdbenv.PeerDBClickHouseInitialLoadPartsPerPartition(ctx, s.config.Env)
+	if err != nil {
+		s.logger.Warn("failed to get chunking parts, proceeding without chunking", slog.Any("error", err))
+		numParts = 1
+	}
+	numParts = max(numParts, 1)
+
+	for i := range numParts {
+		var whereClause string
+		if numParts > 1 {
+			whereClause = fmt.Sprintf(" WHERE cityHash64(`%s`) %% %d = %d", hashColName, numParts, i)
+		}
+		query := fmt.Sprintf(
+			"INSERT INTO `%s`(%s) SELECT %s FROM s3('%s','%s','%s'%s, 'Avro')%s SETTINGS throw_on_max_partitions_per_insert_block = 0",
```

Comment:
```suggestion
			"INSERT INTO `%s`(%s) SELECT %s FROM s3('%s','%s','%s'%s, 'Avro')%s",
```
don't think we want to add that setting here